sites/website/src/docs/design-systems/creating-a-component-library.md

```diff
@@ -0,0 +1,117 @@
+
+```ts
+import { html, css  } from "@microsoft/fast-element";
+import { ElementDefinitionContext, FoundationElementDefinition } from "@microsoft/fast-foundation";
+
+const myElementStyles = (
+context: ElementDefinitionContext,
+definition: FoundationElementDefinition)
+ => css`/* ... */`;
+const myElementTemplate = (
+context: ElementDefinitionContext,
+definition: FoundationElementDefinition
+) => html`<!-- ... -->`;
+```
+
+Let’s take a closer look at what these two arguments give us.
+
+#### What is the `ElementDefinitionContext`?
+// TODO Rob - I'm not sure how to explain this argument well or what all of it's capabilities are
+
+Importantly, the ElementDefinitionContext can be used to inspect the tag-name for other components using the `ElementDefinitionContext.tagFor()` method. This allows usage of other library components in a template or stylesheet without knowing ahead of time what the tag-name for the element will be:
+
+```ts
+import { html } from "@microsoft/fast-element";
+import { FASTButton } from "@microsoft/fast-components";
+import type { ElementDefinitionContext } from "@microsoft/fast-foundation";
+
+const template = (context: ElementDefinitionContext) => {
+	const buttonTag = context.tagFor(FASTButton);
+
+	return html`<${buttonTag}>Click me!</${buttonTag}>`
+}
+```
+
+#### What is the `FoundationElementDefinition`?
+The `FoundationElementDefinition` is the configuration for which the element was defined, allowing inspection of the aggregated options configuration for the component.
+
+This definition can be extended arbitrarily by components during registration, which is what facilitates default slotted content overrides and arbitrary element configuration:
+
+```ts
+import { html } from "@microsoft/fast-element";
+import { ElementDefinitionContext, FoundationElementDefinition } from "@microsoft/fast-foundation";
+
+interface MyElementDefinition extends FoundationElementDefinition {
+    defaultContent?: string;
+}
+
+const template = (context: ElementDefinitionContext, definition: MyElementDefinition) => {
+    return html`<slot>${definition.defaultContent}</slot>`
+}
+```
+
+During component registration with the Design System, an application author can provide a `defaultContent` field as a configuration, which is used by the template if it exists.
+
+### Compose and Export Registration
+Lastly, create and export the registration function so that application authors can register the component in their Design System:
+
+```ts
+export const myElementRegistration = MyElement.compose({
+  baseName: ‘my-element’,
+  template,
+  styles,
+  defaultContent: "Click me!"
+})
+```
+
+Note that the registration is composed with a "defaultContent" value. This value can be provided during composition or as a conditional in the template; either approach will work.
```

Comment:
I don't get neither `during composition` nor `as a conditional in the template`

the example below `myElementRegistration({defaultContent: "Don’t click me!"})` was expected, but I can't relate it to any of these 2 approaches

Reply:
I reworked a good bit of this section. Have a read through the article again and let me know if things are any clearer.